tests/token_authority.tests.ts

```diff
@@ -0,0 +1,623 @@
+import { BN } from "bn.js";
+import { ProgramTestContext } from "solana-bankrun";
+import { deserializeMetadata } from "@metaplex-foundation/mpl-token-metadata";
+import {
+  BaseFee,
+  ConfigParameters,
+  createConfig,
+  CreateConfigParams,
+  createPoolWithSplToken,
+  createPoolWithToken2022,
+} from "./instructions";
+import { VirtualCurveProgram } from "./utils/types";
+import { Keypair, LAMPORTS_PER_SOL, PublicKey } from "@solana/web3.js";
+import { deriveMetadataAccount, fundSol, startTest } from "./utils";
+import {
+  createVirtualCurveProgram,
+  MAX_SQRT_PRICE,
+  MIN_SQRT_PRICE,
+  U64_MAX,
+} from "./utils";
+import { getVirtualPool } from "./utils/fetcher";
+import {
+  ACCOUNT_SIZE,
+  ACCOUNT_TYPE_SIZE,
+  ExtensionType,
+  getExtensionData,
+  MetadataPointerLayout,
+  MintLayout,
+  NATIVE_MINT,
+} from "@solana/spl-token";
+import { expect } from "chai";
+
+describe("Create pool with token2022", () => {
+  let context: ProgramTestContext;
+  let admin: Keypair;
+  let operator: Keypair;
+  let partner: Keypair;
+  let user: Keypair;
+  let poolCreator: Keypair;
+  let program: VirtualCurveProgram;
+  // token2022 pool
+  let creatorUpdateAuthorityToken2022Config: PublicKey;
+  let immutableToken2022Config: PublicKey;
+  let partnerUpdateAuthorityToken2022Config: PublicKey;
+  let creatorUpdateAndMintAuthorityToken2022Config: PublicKey;
+  let partnerUpdateAndMintAuthorityToken2022Config: PublicKey;
+  // spl pool
+  let creatorUpdateAuthoritySplConfig: PublicKey;
+  let immutableSplConfig: PublicKey;
+  let partnerUpdateAuthoritySplConfig: PublicKey;
+  let creatorUpdateAndMintAuthoritySplConfig: PublicKey;
+  let partnerUpdateAndMintAuthoritySplConfig: PublicKey;
+
+  before(async () => {
+    context = await startTest();
+    admin = context.payer;
+    operator = Keypair.generate();
+    partner = Keypair.generate();
+    user = Keypair.generate();
+    poolCreator = Keypair.generate();
+    const receivers = [
+      operator.publicKey,
+      partner.publicKey,
+      user.publicKey,
+      poolCreator.publicKey,
+    ];
+    await fundSol(context.banksClient, admin, receivers);
+    program = createVirtualCurveProgram();
+  });
+
+  it("Partner create config", async () => {
```

Comment:
10 tests:
token2022:
- 5 cases for token authority
spl-token:
- 5 cases for token authority

Reply:
added